File: engine/python/jsonschema_llm_engine/exceptions.py
"""Exception hierarchy for the jsonschema-llm Python engine."""


class EngineError(RuntimeError):
    """Base exception for unrecoverable engine failures."""


class SchemaConversionError(EngineError):
    """WASI convert operation failed."""


class RehydrationError(EngineError):
    """WASI rehydrate operation failed."""


class ResponseParsingError(EngineError):
    """Formatter couldn't parse/extract content from LLM response."""

File: engine/ruby/lib/json_schema_llm_engine/exceptions.rb
# frozen_string_literal: true

# Exception hierarchy for the json-schema-llm Ruby engine.

module JsonSchemaLlmEngine
  # Base error for all engine failures.
  class EngineError < StandardError; end

  # Raised when schema conversion via WASI fails.
  class SchemaConversionError < EngineError; end

  # Raised when rehydration via WASI fails.
  class RehydrationError < EngineError; end

  # Raised when parsing an LLM response fails.
  class ResponseParsingError < EngineError; end
end
